### PR TITLE
DiscoveryEngineDataStore: mockgcp support

### PR DESCRIPTION
**Pull request by justinsb**
- **DiscoveryEngineDataStore: mockgcp support**
- **mockgcp: generate code for discoveryengine**
- **DiscoveryEngineDataStore: protos for mockgcp**
- **DiscoveryEngineDataStore: golden output for mockgcp**

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/discoveryengine/v1alpha1/discoveryenginedatastore/discoveryenginedatastoreminimal/_http.log [hunk 1]
```diff
@@ -56,8 +56,7 @@ X-Xss-Protection: 0
     "defaultSchemaId": "default_schema",
     "displayName": "My first data store",
     "industryVertical": 1,
-    "name": "projects/${projectNumber}/locations/global/collections/default_collection/dataStores/datastore-${uniqueId}",
-    "servingConfigDataStore": {}
```
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/discoveryengine/v1alpha1/discoveryenginedatastore/discoveryenginedatastoreminimal/_http.log:60]: I think we want this `servingConfigDataStore` field showing in the GCP object.

**Review by yuwenma — COMMENTED** [review]
/approve
/lgtm

One nit, not a blocker

File: pkg/test/resourcefixture/testdata/basic/discoveryengine/v1alpha1/discoveryenginedatastore/discoveryenginedatastoreminimal/_http.log (hunk 1 above)
Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/discoveryengine/v1alpha1/discoveryenginedatastore/discoveryenginedatastoreminimal/_http.log:60]: It's not in the protos yet

**Pull request merged**